temporalio/converter.py

```diff
@@ -15,11 +18,11 @@
 import temporalio.api.common.v1
 
 
-class DataConverter(ABC):
-    """Base converter to/from multiple payloads/values."""
+class PayloadConverter(ABC):
```

Comment:
I missed this in the first review, any reason you prefer an ABC over runtime checkable protocol?

Reply:
Python seems to greatly favor abstract classes over protocols for many reasons. One obvious reason is that protocols were only introduced in 3.8 (though of course we can use `typing_extensions`).

Python rarely uses protocols and we should probably rarely use them too unless we have a really good reason for duck typing. I don't think this is a good enough reason.

Reply:
I think ABC was preferred before protocols were introduced.
Don't have a strong preference here. I tend to not force inheritance if it's not strictly required.

Reply:
> I think ABC was preferred before protocols were introduced.

Agreed, was mostly just `raise NotImplementedError` functions or default impls

> I tend to not force inheritance if it's not strictly required.

I usually agree. I can change this if we really want, but the idea of protocol use in Python just doesn't seem that prevalent.

Reply:
I think ABC is going to be simpler for most users.